packages/preferences/src/browser/preference-service.spec.ts

```diff
@@ -68,16 +68,22 @@ before(async () => {
     testContainer.bind(WorkspacePreferenceProvider).toSelf().inSingletonScope();
 
     testContainer.bind(PreferenceProviderProvider).toFactory(ctx => (scope: PreferenceScope) => {
-        const userProvider = ctx.container.get(UserPreferenceProvider);
-        const workspaceProvider = ctx.container.get(WorkspacePreferenceProvider);
-
-        sinon.stub(userProvider, 'onDidPreferencesChanged').get(() =>
-            mockUserPreferenceEmitter.event
-        );
-        sinon.stub(workspaceProvider, 'onDidPreferencesChanged').get(() =>
-            mockWorkspacePreferenceEmitter.event
-        );
-        return scope === PreferenceScope.User ? userProvider : workspaceProvider;
+        switch (scope) {
+            case PreferenceScope.Default:
+                return ctx.container.get(PreferenceSchemaProvider);
+            case PreferenceScope.User:
+                const userProvider = ctx.container.get(UserPreferenceProvider);
+                sinon.stub(userProvider, 'onDidPreferencesChanged').get(() =>
+                    mockUserPreferenceEmitter.event
+                );
+                return userProvider;
+            default:
```

Comment:
If we add workspaceScope and FolderScope, we could add a test for anything else and see if ever we try to modify the scope functionality, what do you think ?